### PR TITLE
critical for writing from mutiple threads

### PR DESCRIPTION
**Pull request by theromis**
I think better to protect this calls in library versus writing locks in a lot of local places

### CODE REVIEW

File: recws.go [hunk 1]
```diff
@@ -66,10 +66,10 @@ func (rc *RecConn) setIsConnected(state bool) {
 // Close closes the underlying network connection without
 // sending or waiting for a close frame.
 func (rc *RecConn) Close() {
-	rc.mu.RLock()
-	defer rc.mu.RUnlock()
 	if rc.Conn != nil {
```
Comment (loeffel-io) [inline recws.go]: should have a reading lock

File: recws.go [hunk 2]
```diff
@@ -66,10 +66,10 @@ func (rc *RecConn) setIsConnected(state bool) {
...
+		rc.mu.RLock()
```
Comment (loeffel-io) [inline recws.go]: should be a lock instead of a reading lock

**Comment by loeffel-io** [comment]
I think this could be the solution: 

```go
func (rc *RecConn) getConn() *websocket.Conn {
	rc.mu.RLock()
	defer rc.mu.RUnlock()

	return rc.Conn
}

// Close closes the underlying network connection without
// sending or waiting for a close frame.
func (rc *RecConn) Close() {
	if rc.getConn() != nil {
		rc.mu.Lock()
		rc.Conn.Close()
		rc.mu.Unlock()
	}

	rc.setIsConnected(false)
}
```

**Comment by loeffel-io** [comment]
I will test this in a long time running process for the next 24 hours 
https://github.com/loeffel-io/recws/pull/1/files

**Comment by theromis** [comment]
Yes, this solution should work. Let's see the test results.

**Comment by loeffel-io** [comment]
Looks pretty good! 
@mariuspass

**Comment by theromis** [comment]
@loeffel-io done, please check

**Comment by loeffel-io** [comment]
Looks good! 
@mariuspass could you please merge?

**Pull request merged**